### PR TITLE
Added AWS Service Discovery block to aws_ecs_service

### PR DESCRIPTION
**Pull request by larstobi**

### CODE REVIEW

**Comment by mikael-lindstrom** [comment]
This would make service discovery required right?

**Comment by larstobi** [comment]
I fixed that by using a dynamic block @mikael-lindstrom

File: variables.tf [hunk 1]
```diff
@@ -137,3 +137,8 @@ variable "repository_credentials_kms_key" {
   description = "key id, key ARN, alias name or alias ARN of the key that encrypted the repository credentials"
   type        = string
 }
+
+variable "service_registry_arn" {
+  default = null
```
Comment (itsdalmo) [inline variables.tf]: The default value for a string can just be `""`.

File: variables.tf [hunk 2]
```diff
@@ -137,3 +137,8 @@ variable "repository_credentials_kms_key" {
   description = "key id, key ARN, alias name or alias ARN of the key that encrypted the repository credentials"
   type        = string
 }
+
+variable "service_registry_arn" {
```
Comment (itsdalmo) [inline variables.tf]: Need to add a `description` for the new variable.

File: main.tf [hunk 3]
```diff
@@ -190,6 +190,17 @@ resource "aws_ecs_service" "service" {
...
+  dynamic "service_registries" {
+    for_each = var.service_registry_arn == null ? [] : [1]
+    content {
+      registry_arn   = var.service_registry_arn
+      container_port = var.task_container_port
+      container_name = var.container_name != "" ? var.container_name : var.name_prefix
+    }
+  }
+
+  lifecycle { create_before_destroy = true }
```
Comment (itsdalmo) [inline main.tf]: This seems wrong and not related to the PR?

File: variables.tf (hunk 1 above)
Comment (larstobi) [reply variables.tf]: I think that null is more correct here.

Comment (larstobi) [reply variables.tf]: I could be wrong...

**Comment by larstobi** [comment]
I don't think multiple service registries is supported. And I also saw an issue that it threw an error with multiple service registries.

**Pull request merged**